specification/data-semantic-conventions.md

```diff
@@ -249,70 +279,118 @@ but due to `http.scheme`, `http.host` and `http.target` being set, it would be r
 As explained above, these separate values are preferred but if for some reason the URL is available but the other values are not,
 URL can replace `http.scheme`, `http.host` and `http.target`.
 
-## Databases client calls
+## Databases (SQL and NoSQL) client calls
+
+> 🚧 WORK IN PROGRESS
 
 For database client call the `SpanKind` MUST be `Client`.
 
 Span `name` should be set to low cardinality value representing the statement
-executed on the database. It may be stored procedure name (without argument), sql
+executed on the database. It may be stored procedure name (without argument), SQL
 statement without variable arguments, etc. When it's impossible to get any
 meaningful representation of the span `name`, it can be populated using the same
 value as `db.instance`.
 
-Note, Redis, Cassandra, HBase and other storage systems may reuse the same
-attribute names.
+### Connection-level attributes
+
+These attributes will usually be the same for all operations performed over the same database connection
+although some database systems may allow to switch to a different `db.user` within the same connection
+and other database systems may not even have the concept of a connection.
 
 | Attribute name | Notes and examples                                           | Required? |
 | :------------- | :----------------------------------------------------------- | --------- |
-| `component`    | Database driver name or database name (when known) `"JDBI"`, `"jdbc"`, `"odbc"`, `"postgreSQL"`. | Yes       |
-| `db.type`      | Database type. For any SQL database, `"sql"`. For others, the lower-case database category, e.g. `"cassandra"`, `"hbase"`, or `"redis"`. | Yes       |
-| `db.instance`  | Database instance name. E.g., In java, if the jdbc.url=`"jdbc:mysql://db.example.com:3306/customers"`, the instance name is `"customers"`. | Yes       |
-| `db.statement` | A database statement for the given database type. Note, that the value may be sanitized to exclude sensitive information. E.g., for `db.type="sql"`, `"SELECT * FROM wuser_table"`; for `db.type="redis"`, `"SET mykey 'WuValue'"`. | Yes       |
+| `component` | Always the string `"db"` | Yes       |
+| `db.type` | The lower-case database category, e.g. `"sql"`, `"cassandra"`, `"hbase"`, or `"redis"`. | Yes       |
```

Comment:
IMHO `"sql"`, `"cassandra"`, `"hbase"`, or `"redis"` is not consistent. While "`sql"` stands for all relational databases, `"cassandra"`, `"hbase"`, or `"redis"` are concrete database technologies.

Thinking about it more, even `"sql"` is not a database-type, it only declares the query language. Correct type would be "Relational Database".

IMHO these would be proper database "types":

**Key-Value-Database** (e.g. redis, couchbase, dynamo, ignite, ...)
https://en.wikipedia.org/wiki/Key-value_database

**Relational Database** (e.g. postgres, mysql, sqlserver, oracledb, ...)
https://en.wikipedia.org/wiki/Relational_database

**Document-oriented Database** (e.g. couchdb, mongodb, ...)
https://en.wikipedia.org/wiki/Document-oriented_database

**Wide Column Store** (e.g. cassanda, hbase, ...)
https://en.wikipedia.org/wiki/Wide_column_store

**Graph database** (e.g. neo4j, ...)
https://en.wikipedia.org/wiki/Graph_database


This list is probably non-exhaustive.

Reply:
I think the granularity is correct here:
* SQL: All databases that adhere to SQL concepts. 
* Redis: All databases that adhere to redis concepts (only redis AFAIK)
* Cassandra: All databases that adhere to the cassandra concepts (only cassandra; e.g. CQL, etc)

SQL is a standard that DBs choose to adhere to (not 100% of course, but still). On the other hand, the types you mentioned are merely post-hoc descriptions of databases and the distinctions are not clear-cut.